compiler/qsc_eval/src/lib.rs

```diff
@@ -1972,5 +1996,25 @@ fn is_counting_call(name: &str) -> bool {
             | "StopCountingOperation"
             | "StartCountingFunction"
             | "StopCountingFunction"
+            | "StartCountingQubits"
+            | "StopCountingQubits"
     )
 }
+
+#[derive(Default)]
+struct QubitCounter {
+    seen: FxHashSet<usize>,
+    count: i64,
+}
+
+impl QubitCounter {
+    fn allocated(&mut self, qubit: usize) {
+        if self.seen.insert(qubit) {
```

Comment:
Does this mean qubit reuse still counts as the one qubit (which makes sense).

Reply:
Yes, that's correct. If qubit A is allocated, used, released, then allocated again, that still counts as just one. I added some unit tests to make sure of this.